infrastructure/sentry-subgraph/src/esXai.ts

```diff
@@ -0,0 +1,87 @@
+import {  BigInt, Bytes, ethereum, log } from "@graphprotocol/graph-ts"
+import { 
+    RedemptionStarted as RedemptionStartedEvent,
+    RedemptionCancelled as RedemptionCancelledEvent,
+    RedemptionCompleted as RedemptionCompletedEvent,
+    VoucherIssued as VoucherIssuedEvent
+} from "../generated/esXai/esXai";
+import { RedemptionRequest } from "../generated/schema";
+
+export function handleRedemptionStarted(event: RedemptionStartedEvent): void {
+
+    // Create a new redemption request entity 
+    const request = new RedemptionRequest(event.params.user.toHexString() + "_"  + event.params.index.toString());
+    
+    // Decode the transaction input data to get the redemption amount and duration
+    const inputData = event.transaction.input.toHexString().slice(10); // Remove function selector
+    const decoded = ethereum.decode('(uint256,uint256)', Bytes.fromHexString(inputData) as Bytes);
+
+    if (decoded) {      
+        const decodedTuple = decoded.toTuple();
+            request.amount = decodedTuple[0].toBigInt();
+            request.duration = decodedTuple[1].toBigInt();
+    } else {
+        log.warning("Failed to decode startRedemption TX: {}", [event.transaction.hash.toHexString()]);
+        return;
+    }
+    request.sentryWallet = event.params.user.toHexString();
```

Comment:
Will this just work even if the user does not exists as a sentry wallet yet ?

Reply:
No, but a Sentry Wallet will be created when the user receives the esXai in the transfer handler.